vali_objects/utils/position_manager.py

```diff
@@ -1299,3 +1299,16 @@ def generate_correction_templates(self):
             f.write(str(x) + '\n')
 
         f.close()
+
+    def calculate_net_portfolio_leverage(self, hotkey):
+        """
+        Calculate leverage across all open positions
+        Normalize each asset class with a multiplier
+        """
+        positions = self.get_all_miner_positions(hotkey, only_open_positions=True)
+
+        portfolio_leverage = 0.0
+        for position in positions:
+            portfolio_leverage += position.get_net_leverage() * position.trade_pair.leverage_multiplier
```

Comment:
need to to abs value since some positions could be short (negative) and some long

Reply:
Good catch, thanks!